gta-gallery/src/components/editUser/index.jsx

```diff
@@ -1,57 +1,87 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { useHistory } from 'react-router-dom';
-import { InputFields } from '../inputFields';
 import { Button } from '../button';
-import { editUser } from '../../services/userService';
+import { editUser, getCurrentUser } from '../../services/userService';
+import { Form, Input } from 'antd';
+import { setUser } from '../../store/auth/actions';
+import { useDispatch } from 'react-redux';
 
 import './styles.scss';
 
 export const EditUser = () => {
   const history = useHistory();
+  const dispatch = useDispatch();
 
+  const [form] = Form.useForm();
   const [firstNameValue, setFirstNameValue] = useState('');
   const [lastNameValue, setLastNameValue] = useState('');
   const [usernameValue, setUsernameValue] = useState('');
 
-  const inputFields = [
-    {
-      name: 'firstName',
-      label: 'First name',
-      onChange: setFirstNameValue,
-      itemValue: firstNameValue,
-    },
-    {
-      name: 'lastName',
-      label: 'Last name',
-      onChange: setLastNameValue,
-      itemValue: lastNameValue,
-    },
-    {
-      name: 'username',
-      label: 'Username',
-      onChange: setUsernameValue,
-      itemValue: usernameValue,
-    },
-  ];
+  const getData = async () => {
+    const data = await getCurrentUser();
+    if (!data) return;
+    setFirstNameValue(data.firstName);
+    setLastNameValue(data.lastName);
+    setUsernameValue(data.username);
+    form.resetFields();
+  };
 
   const handleEditUser = async () => {
-    if (usernameValue && firstNameValue && lastNameValue) {
-      await editUser(firstNameValue, lastNameValue, usernameValue);
-      history.push('/');
-    }
-    return;
+    await editUser(firstNameValue, lastNameValue, usernameValue);
+    dispatch(
+      setUser({
+        firstName: firstNameValue,
+        lastName: lastNameValue,
+        username: usernameValue,
+      })
+    );
+    history.push('/');
   };
 
+  useEffect(() => {
+    getData();
+  }, []);
+
   return (
     <div className='edit-user-container'>
       <div className='edit-user'>
         <h3 className='edit-user-header'>Edit User</h3>
-        <div>
-          <InputFields fieldsArray={inputFields} />
-        </div>
-        <div className='edit-user-button'>
-          <Button onClick={handleEditUser} text='Edit user' />
-        </div>
+        <Form
+          form={form}
+          initialValues={{
+            firstName: firstNameValue,
+            lastName: lastNameValue,
+            username: usernameValue,
+          }}
+        >
+          <Form.Item label='First name' name='firstName'>
+            <Input
+              type='text'
+              onChange={(e) => setFirstNameValue(e.target.value)}
```

Comment:
in changePassword you have written this in another way

Reply:
I know.